auditlog/audit_log_blackbox_test.go

```diff
@@ -0,0 +1,205 @@
+package auditlog_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/errors"
+
+	"github.com/fabric8-services/admin-console/auditlog"
+	"github.com/fabric8-services/admin-console/configuration"
+	"github.com/fabric8-services/fabric8-common/resource"
+	testsuite "github.com/fabric8-services/fabric8-common/test/suite"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RepositoryBlackBoxTest struct {
+	testsuite.DBTestSuite
+	repo auditlog.Repository
+}
+
+func TestRecordRepository(t *testing.T) {
+	resource.Require(t, resource.Database)
+	config, err := configuration.New()
+	require.NoError(t, err)
+	suite.Run(t, &RepositoryBlackBoxTest{DBTestSuite: testsuite.NewDBTestSuite(config)})
+}
+
+func (s *RepositoryBlackBoxTest) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.repo = auditlog.NewRepository(s.DB)
+}
+
+func (s *RepositoryBlackBoxTest) TestCreateRecord() {
+
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		before := time.Now()
+		auditLog := auditlog.AuditLog{
+			EventTypeID: auditlog.UserSearch,
+			IdentityID:  uuid.NewV4(),
+			EventParams: auditlog.EventParams{},
+		}
+		// when
+		err := s.repo.Create(context.Background(), &auditLog)
+		// then
+		require.NoError(t, err)
+		assert.NotEqual(t, uuid.NullUUID{}, auditLog.ID)
+		assert.True(t, auditLog.CreatedAt.After(before)) // "is after before". hahahaha....
+	})
+
+	s.T().Run("failure", func(t *testing.T) {
+
+		t.Run("missing event type", func(t *testing.T) {
+			// given
+			auditLog := auditlog.AuditLog{
+				IdentityID:  uuid.NewV4(),
+				EventParams: auditlog.EventParams{},
+			}
+			// when
+			err := s.repo.Create(context.Background(), &auditLog)
+			// then
+			require.Error(t, err)
+		})
+
+		t.Run("missing identity id", func(t *testing.T) {
+			// given
+			auditLog := auditlog.AuditLog{
+				EventTypeID: auditlog.UserSearch,
+				EventParams: auditlog.EventParams{},
+			}
+			// when
+			err := s.repo.Create(context.Background(), &auditLog)
+			// then
+			require.Error(t, err)
```

Comment:
Check error type and message too.

Reply:
ok

Reply:
done in 111ae7eb7d9978c56dbf97c9e2448578d23e18b6